esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst

```diff
@@ -0,0 +1,258 @@
+Sonoff Fish Pond Pump
+=====================
+
+.. seo::
+    :description: Firmware for an Sonoff S2X smart switch 
+    :image: images/sonoff-s2x.jpg
+    :keywords: sonoff, esp8266, home assistant, smart socket
+
+.. figure:: images/sonoff-s2x.jpg
+    :align: center
+    :width: 30.0%
+
+This is a basic firmware for the Sonoff S2X smart switch series. The features are:
+ 
+* The switch is automatically registered to HA via MQTT.
+* A *single click* toggles the relay (works offline).
+* A *double click*, a *triple click*, or a *longpress* triggers custom actions on HA. 
+
+.. warning::
+
+    Do NOT connect your device to the mains when programming it.
+    Take care working with the mains voltage at all times!
+
+1. Preparation
+--------------
+Prepare your socket according to these instructions 
+
+- :doc:`Sonoff S20 </esphomeyaml/devices/sonoff_s20>`
+- `Sonoff S26 <https://github.com/arendst/Sonoff-Tasmota/wiki/Sonoff-S26-Smart-Socket/>`
+
+2. ESPHome Configuration
+------------------------
+
+Here is the configuration with the basic operations outlined above.
+
+.. code-block:: yaml
+    substitutions:
+      devicename: sonoff1
+
+    esphomeyaml:
+      name: $devicename
+      platform: ESP8266
+      board: esp01_1m
+      board_flash_mode: dout
+
+
+    wifi:
+      networks:
+        - ssid: !secret wifi_ssid
+          password: !secret wifi_password
+        - ssid: !secret wifi_ssid2
+          password: !secret wifi_password2
+    # power_save_mode: none
+      ap:
+        ssid: "$devicename AP"
+        password: !secret  ap_password 
+        channel: 10
```

Comment:
Also remove channel. The example configs here should be as simple as possible (possibly even remove `ap:`)